app/models/product_reading.rb

```diff
@@ -91,7 +91,7 @@ class ProductReading < Ekylibre::Record::Base
   end
 
   after_save do
-    last_reading = product.readings.where(indicator_name: indicator_name).last
+    last_reading = ProductReading.where(product_id: product.id, indicator_name: indicator_name).last
```

Comment:
last_reading can still be Nil here (no ProductReading)

Reply:
i let you see what i did to fix the issue